crates/api_common/src/site.rs

```diff
@@ -151,6 +175,29 @@ pub struct EditSite {
   pub application_email_admins: Option<bool>,
   pub hide_modlog_mod_names: Option<bool>,
   pub discussion_languages: Option<Vec<LanguageId>>,
+  pub slur_filter_regex: Option<String>,
+  pub actor_name_max_length: Option<i32>,
+  pub rate_limit_message: Option<i32>,
+  pub rate_limit_message_per_second: Option<i32>,
+  pub rate_limit_post: Option<i32>,
+  pub rate_limit_post_per_second: Option<i32>,
+  pub rate_limit_register: Option<i32>,
+  pub rate_limit_register_per_second: Option<i32>,
+  pub rate_limit_image: Option<i32>,
+  pub rate_limit_image_per_second: Option<i32>,
+  pub rate_limit_comment: Option<i32>,
+  pub rate_limit_comment_per_second: Option<i32>,
+  pub rate_limit_search: Option<i32>,
+  pub rate_limit_search_per_second: Option<i32>,
+  pub federation_enabled: Option<bool>,
+  pub federation_debug: Option<bool>,
+  pub federation_strict_allowlist: Option<bool>,
+  pub federation_http_fetch_retry_limit: Option<i32>,
+  pub federation_worker_count: Option<i32>,
+  pub captcha_enabled: Option<bool>,
+  pub captcha_difficulty: Option<String>,
+  pub allowed_instances: Option<Vec<String>>,
+  pub blocked_instances: Option<Vec<String>>,
   pub auth: Sensitive<String>,
```

Comment:
I think its worth to move some of these fields into nested structs, at least rate limit so it doesnt have to be specified and kept in sync in two places.

Reply:
Which two places do you mean. The API structs have to be distinct from the DB ones.